server/server_test.go

```diff
@@ -14,45 +18,112 @@ import (
 var _ = Describe("rode server", func() {
 	var (
 		rodeServer    pb.RodeServer
-		grafeasClient *mockGrafeasClient
+		grafeasClient *mocks.MockGrafeasV1Beta1Client
+		mockCtrl      *gomock.Controller
 	)
 
 	BeforeEach(func() {
-		grafeasClient = &mockGrafeasClient{}
+		mockCtrl = gomock.NewController(GinkgoT())
+		grafeasClient = mocks.NewMockGrafeasV1Beta1Client(mockCtrl)
 		rodeServer = NewRodeServer(logger.Named("rode server test"), grafeasClient)
 	})
 
+	AfterEach(func() {
+		mockCtrl.Finish()
+	})
+
 	When("occurrences are created", func() {
 		var (
 			randomOccurrence *grafeas_proto.Occurrence
 		)
 
 		JustBeforeEach(func() {
 			randomOccurrence = createRandomUnspecifiedOccurrence()
-			grafeasClient.preparedBatchCreateOccurrenceResponse = &grafeas_proto.BatchCreateOccurrencesResponse{
+		})
+
+		It("should forward the batch create occurrence request to grafeas", func() {
+			// expected Grafeas BatchCreateOccurrences request
+			grafeasBatchCreateOccurrencesRequest := &grafeas_proto.BatchCreateOccurrencesRequest{
+				Parent: "projects/rode",
 				Occurrences: []*grafeas_proto.Occurrence{
 					randomOccurrence,
 				},
 			}
-		})
+			// mocked Grafeas BatchCreateOccurrences response
+			grafeasBatchCreateOccurrencesResponse := &grafeas_proto.BatchCreateOccurrencesResponse{
+				Occurrences: []*grafeas_proto.Occurrence{
+					randomOccurrence,
+				},
+			}
+			// ensure Grafeas BatchCreateOccurrences is called with expected request and inject response
+			grafeasClient.EXPECT().BatchCreateOccurrences(gomock.AssignableToTypeOf(context.Background()), gomock.Eq(grafeasBatchCreateOccurrencesRequest)).Return(grafeasBatchCreateOccurrencesResponse, nil)
```

Comment:
is there a way to pre-configure this response in a `BeforeEach`?  we usually try to capture these setup steps there and leave the `It` for the actual assertions.

Reply:
I moved the initialization of the request and response objects into a `BeforeEach` block which makes sense. I'm not as sure about the mock itself. I don't think we can separate asserting the method was called from injecting the response so i think it makes sense to leave it in the `It` block but i'm curious what you think?